### ISSUE TITLE
after updating nightly version cmdheight is broken

### ISSUE THREAD

**Issue by tigerkiller55891** [neovim/neovim#20243]
### Neovim version (nvim -v)

nvim v0.8.0-dev-1112-g6d557e324

### Vim (not Nvim) behaves the same?

i don't know

### Operating system/version

Arch linux 5.19.9-zen1-1-zen

### Terminal name/version

alacritty 0.10.1

### $TERM environment variable

alacritty

### Installation

AUR

### How to reproduce the issue

install nightly version 
set cmdheight to 0 in lua
open any file and enter command mode

### Expected behavior

for the command mode to work normally

### Actual behavior

![neovim-broke](https://user-images.githubusercontent.com/80734842/190934261-5d2548a5-30db-4677-9374-3ee418368155.png)

**Comment by tigerkiller55891** [comment]
my config files are here if they will help https://gitlab.com/mystuff15/nvim-mess/-/tree/master/lua/user/lsp

**Comment by zeertzjq** [comment]
May be the same problem as #20106.

**Comment by zeertzjq** [comment]
Like #20106, removing this line also fixes the floating window redrawing problem:
```diff
diff --git a/src/nvim/drawscreen.c b/src/nvim/drawscreen.c
index 957e8ae4d..688e98125 100644
--- a/src/nvim/drawscreen.c
+++ b/src/nvim/drawscreen.c
@@ -447,7 +447,6 @@ int update_screen(int type)
       msg_grid.throttled = false;
       // UPD_CLEAR is already handled
       if (type == UPD_NOT_VALID && !ui_has(kUIMultigrid) && msg_scrolled) {
-        ui_comp_set_screen_valid(false);
         for (int i = valid; i < Rows - p_ch; i++) {
           grid_clear_line(&default_grid, default_grid.line_offset[i],
                           Columns, false);
```

Not sure about the command line redraw.

**Comment by zeertzjq** [comment]
It seems that `:redrawstatus` in cmdline mode with cmdheight=0 will cause command line to be hidden.

**Comment by adelarsq** [comment]
For now to avoid this bug you can use `cmdheight=1` or disable plugins that change cmd behaviour.

**Comment by Shougo** [comment]
https://github.com/neovim/neovim/pull/20249 fixes the problem.

**Comment by zeertzjq** [comment]
The float window redraw problem has been fixed by #20249 
The cmdline redraw problem is tracked in #20281
